nomenclature/config.py

```diff
@@ -0,0 +1,43 @@
+from pathlib import Path
+from typing import Dict
+from pydantic import BaseModel, validator
+
+import yaml
+
+
+class DataStructureConfig(BaseModel):
+    """A class for configuration of a DataStructureDefinition
+
+    Attributes
+    ----------
+    config : dict
+        Attributes for configuring the DataStructureDefinition
+
+    """
+
+    config: Dict[str, Dict] = {}
+
+    def get(self, key):
+        return self.config.get(key)
+
+    @classmethod
+    def from_file(cls, path: Path, file: str, dimensions: list):
+        """Read a DataStructureConfig from a file
+
+        Parameters
+        ----------
+        path : :class:`pathlib.Path` or path-like
+            `definitions` directory
+        file : str
+            File name
+        dimensions : list
+            List of valid dimensions
+
+        """
+        with open(path / file, "r", encoding="utf-8") as stream:
+            config = yaml.safe_load(stream)
+
+            if invalid_config := [i for i in config if i not in dimensions]:
```

Comment:
If you explicitly list all the dimensions in `DataStructureConfig` and set the config to not allow additional attributes you probably don't need the `dimensions` input parameter and this check.
I would also say that we can remove the `dimensions` input parameter from the `DataStructureDefinition` constructor. Reason being that I haven't come across a use case where we have a code list for a particular dimension but then don't want to validate that dimension. 
Maybe there is a use case though is which case this check is necessary. In any case this would be a new issue anyway.

Reply:
We have such a use-case in openENTRANCE and ARIADNE, where we use the same workflow repo for different database instances - in each case, one instance checks the "scenario" dimension and the other(s) ignore it...

Reply:
Ok valid. I'd still using an iterator other than `i` so that it's easier to comprehend what you're iterating over. Maybe `dimension` would be better.